openmc/data/resonance_covariance.py

```diff
@@ -0,0 +1,841 @@
+
+    def reconstruct(self, energies, resonances, sampleN):
+        """Evaluate the cross section at specified energies for an already
+        sampled set of resonance parameters. 
+    
+        Parameters
+        ----------
+        energies : float or Iterable of float
+            Energies at which the cross section should be evaluated
+        resonances : openmc.data.Resonance object
+            Corresponding resonance range with File 2 data. Used for
+            reconstruction method
+        sampleN : int
+            Index of sample of resonance parameters to be used
+    
+        Returns
+        -------
+        3-tuple of float or numpy.ndarray
+            Elastic, capture, and fission cross sections at the specified
+            energies
+    
+        """
+        if self.samples[sampleN] is None:
+            raise ValueError("Sample of resonance parameters has not been set.")
+        sample_parameters = self.samples[sampleN]
+        xs_array = resonances.reconstruct(energies, use_sample = True,
+                                          sample_parameters = sample_parameters)
+        return xs_array
+
+class MultiLevelBreitWignerCovariance(ResonanceCovarianceRange):
+    """Multi-level Breit-Wigner resolved resonance formalism covariance data.
+    Parameters
+    ----------
+    energy_min : float
+        Minimum energy of the resolved resonance range in eV
+    energy_max : float
+        Maximum energy of the resolved resonance range in eV
+
+    Attributes
+    ----------
+    energy_min : float
+        Minimum energy of the resolved resonance range in eV
+    energy_max : float
+        Maximum energy of the resolved resonance range in eV
+    parameters: pandas.DataFrame
+        Resonance parameters
+    covariance : numpy.array
+        The covariance matrix contained within the ENDF evaluation
+    lcomp : int
+        Flag indicating the format of the covariance matrix within the ENDF file
+    mpar : int
+        Number of parameters in covariance matrix for each individual resonance
+    formalism : str
+        String descriptor of formalism
+    """
+
+    def __init__(self, energy_min, energy_max):
+        super().__init__(energy_min, energy_max)
+        self.parameters = None
+        self.covariance = None
+        self.mpar = None
+        self.lcomp = None
+        self.formalism = 'mlbw'
+            
+    @classmethod
+    def from_endf(cls, ev, file_obj, items, file2params):
+        """Create MLBW covariance data from an ENDF evaluation.
+
+        Parameters
+        ----------
+        ev : openmc.data.endf.Evaluation
+            ENDF evaluation
+        file_obj : file-like object
+            ENDF file positioned at the second record of a resonance range
+            subsection in MF=32, MT=151
+        items : list
+            Items from the CONT record at the start of the resonance range
+            subsection
+        resonances : openmc.data.IncidentNeutron.Resonance object
```

Comment:
I think this should be the dataframe of parameters from a `ResonanceRange` object?

Reply:
Also, needs a description below